jstests/ssl/x509/mkcert.py

```diff
@@ -520,17 +595,43 @@ def process_ecdsa_leaf(cert):
     reqargs = ['openssl', 'req', '-new', '-key', key, '-out', csr, '-subj', subject]
     x509args = ['openssl', 'x509', '-in', csr, '-out', pem, '-req', '-CA', ca, '-CAkey', ca, '-days', '7300', '-sha256', '-set_serial', serial]
     if mode == 'server':
-        reqargs.append('-reqexts=v3_req')
+        reqargs = reqargs + ['-reqexts', 'v3_req']
         extfile = tempfile.mkstemp()[1]
-        open(extfile, 'wt').write('subjectAltName=DNS:localhost,DNS:127.0.0.1')
+        open(extfile, 'wt').write('basicConstraints=CA:FALSE\n')
+        open(extfile, 'at').write('subjectAltName=DNS:localhost,IP:127.0.0.1\n')
+        open(extfile, 'at').write('subjectKeyIdentifier=hash\n')
+        key_usage = ('keyUsage=nonRepudiation,digitalSignature,keyEncipherment\n'
+                     if (cert.get('tags') and "responder" in cert['tags'])
+                     else 'keyUsage=digitalSignature,keyEncipherment\n')
+        open(extfile, 'at').write(key_usage)
+        extended_key_usage = ('extendedKeyUsage=serverAuth,clientAuth,OCSPSigning\n'
+                              if (cert.get('tags') and "responder" in cert['tags'])
+                              else 'extendedKeyUsage=serverAuth,clientAuth\n')
+        open(extfile, 'at').write(extended_key_usage)
+        if cert.get('tags'):
+            if "ocsp" in cert['tags']:
+                if not "responder" in cert['tags']:
+                    open(extfile, 'at').write('authorityInfoAccess = OCSP;URI:http://localhost:9001/power/level/,OCSP;URI:http://localhost:8100/status/\n')
+                    if "must-staple" in cert['tags']:
+                        open(extfile, 'at').write(f'{MUST_STAPLE_KEY_STR}={MUST_STAPLE_VALUE_STR}\n')
         x509args.append('-extfile')
         x509args.append(extfile)
 
     subprocess.check_call(['openssl', 'ecparam', '-name', 'prime256v1', '-genkey', '-out', key])
     subprocess.check_call(reqargs)
     subprocess.check_call(x509args)
 
-    open(dest, 'wt').write(get_header_comment(cert) + "\n" + open(pem, 'rt').read() + open(key, 'rt').read())
+    if "responder" in cert['name']:
```

Comment:
I'm confused as to where you're using the tags field here? Shouldn't it be something like - 
`if 'responder' in cert['tags']:` (with cases for each tag) to reroute the cert for processing from different functions?